lib/win32/certstore/certstore.rb

```diff
@@ -15,30 +16,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-require "win32/certstore/version"
-require 'win32/api/reserved_names'
+require 'forwardable'
 
 module Win32
-  module Certstore
-    include Win32::API::ReservedNames
-    include Chef::Mixin::WideString
+  class Certstore
+    extend Forwardable
 
-    def open store_name

```

Comment:
I think `open` and `close` should remain in `certstore`. Open should be a class method that returns (or yields) a store.

Reply:
I too agree that `open` and `close` methods should be in `certstore`. But @mwrock, if `open` will be a class method that returns a `certstore` object, then that object will contain the certificate handler and users will have access to it.

Reply:
I think the handle can be hidden as a private member. Alternatively you could hide open and close completely. Internally you would have public store operations like `list` check for a handle and call a private instance `open` if there isn't one. For closing, you could use `ObjectSpace.define_finalizer` so that it is automatically freed when it is GC'd. This is the pattern I used in winrm and it worked well: https://github.com/WinRb/WinRM/blob/master/lib/winrm/shells/base.rb#L171